crates/sparrow-runtime/src/prepare.rs

```diff
@@ -426,7 +426,8 @@ fn reader_from_parquet<'a, R: parquet::file::reader::ChunkReader + 'static>(
     let batch_size = get_batch_size(num_rows, num_files);
     let parquet_reader = parquet_reader.with_batch_size(batch_size);
 
-    let raw_metadata = RawMetadata::from_raw_schema(parquet_reader.schema().clone());
+    let raw_metadata = RawMetadata::from_raw_schema(parquet_reader.schema().clone())
+        .change_context_lazy(|| Error::DetermineMetadata)?;
```

Comment:
Should this be `Error::ReadSchema`? same for below

Reply:
same for below

Reply:
i.e. `DetermineMetadata` doesn't make much sense from a user's POV. The cause is from an invalid schema